packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -31,42 +32,23 @@ export const createCart = () =>
         }
 
         try {
-            const guestCartEndpoint = '/rest/V1/guest-carts';
-            const signedInCartEndpoint = '/rest/V1/carts/mine';
-            const cartEndpoint = user.isSignedIn
-                ? signedInCartEndpoint
-                : guestCartEndpoint;
-
-            const cartId = await request(cartEndpoint, {
-                method: 'POST'
-            });
+            const { data, error } = await fetchCartId();
 
-            // write to storage in the background
-            saveCartId(cartId);
-
-            // There is currently an issue in Magento 2
-            // where the first item added to an empty cart for an
-            // authenticated customer gets added with a price of zero.
-            // @see https://github.com/magento/magento2/issues/2991
-            // This workaround is in place until that issue is resolved.
-            if (user.isSignedIn) {
-                await request('/rest/V1/carts/mine/billing-address', {
-                    method: 'POST',
-                    body: JSON.stringify({
-                        address: {},
-                        cartId
-                    })
-                });
+            if (error) {
+                throw new Error('Unable to fetch cartId:', error);
```

Comment:
There's nothing else in the try block that throws, so it seems odd to keep it just to throw inside; get rid of the try/catch and just handle error state (similar to code sample I provided).